### PR TITLE
infoschema: speed up reader for some tables

### PR DESCRIPTION
**Pull request by ywqzzy**
### What problem does this PR solve?

Issue Number: ref #50305 

Problem Summary:

### What changed and how does it work?
Push down some filters for：
1. infoschema.TableStatistics
2. infoschema.TablePartitions
3. infoschema.TableSequences
4. infoschema.TableCheckConstraints
5. infoschema.TableTiDBCheckConstraints
6. infoschema.TableTiDBIndexUsage
7. infoschema.TableTiDBIndexes
8. infoschema.TableViews
9. infoschema.TableConstraints
10. infoschema.TableSchemata

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/planner/core/memtable_predicate_extractor.go [hunk 1]
```diff
@@ -1812,7 +1812,7 @@ func (e *InfoSchemaTablesExtractor) Extract(ctx base.PlanContext,
 	predicates []expression.Expression,
 ) (remained []expression.Expression) {
 	var resultSet, resultSet1 set.StringSet
-	e.colNames = []string{"table_schema", "table_name"}
+	e.colNames = []string{"table_schema", "constraint_schema", "table_name", "constraint_name", "sequence_schema", "sequence_name", "partition_name", "schema_name", "index_name"}
```
Comment (lance6716) [inline pkg/planner/core/memtable_predicate_extractor.go:1815]: the field `InfoSchemaTablesExtractor.colNames` can be replaced by a global constant-like variable?

File: pkg/planner/core/memtable_predicate_extractor.go
Comment (lance6716) [inline pkg/planner/core/memtable_predicate_extractor.go:1]: I see almost all implementations of Extract also can express SkipRequest. Don't know why we move it from implementation's member to a returning value. Returning value has smaller variable scope so less mind burden.

File: pkg/planner/core/memtable_predicate_extractor.go [hunk 2]
```diff
@@ -1812,7 +1812,7 @@ func (e *InfoSchemaTablesExtractor) Extract(ctx base.PlanContext,
...
 	e.ColPredicates = make(map[string]set.StringSet)
 	remained = predicates
 	for _, colName := range e.colNames {
```
Comment (lance6716) [inline pkg/planner/core/memtable_predicate_extractor.go:1818]: maybe batch all `colNames` into one call to reduce overhead. In the new code there are many colNames than before

**Review by lance6716 — COMMENTED** [review]
1 / 5 files viewed

File: pkg/planner/core/memtable_predicate_extractor.go (hunk 2 above)
Comment (ywqzzy) [reply pkg/planner/core/memtable_predicate_extractor.go:1818]: complicated to do so

File: pkg/planner/core/memtable_predicate_extractor.go
Comment (ywqzzy) [reply pkg/planner/core/memtable_predicate_extractor.go:1]: We use it in reader

**Review by tangenta — APPROVED** [review]

**Review by GMHDBJD — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**